go-controller/pkg/node/gateway_iptables.go

```diff
@@ -356,13 +390,20 @@ func getGatewayIPTRules(service *kapi.Service, hasLocalHostEndpoint bool) []iptR
 				continue
 			}
 			if !hasLocalHostEndpoint {
-				for _, clusterIP := range clusterIPs {
-					rules = append(rules, getNodePortIPTRules(svcPort, clusterIP, svcPort.Port)...)
+				if etpLocalNonHostRules {
```

Comment:
```			if !hasLocalHostEndpoint {
				for _, clusterIP := range clusterIPs {
					rules = append(rules, getNodePortIPTRules(svcPort, clusterIP, svcPort.Port)...)
				if etpLocalNonHostRules {
```

hasLocalHostEndpoint and etpLocalNonHostRules seem like overloaded variables that imply gateway mode, if the host has ovn-networked endpoints local to it, and if the policy has ETP=local. I'd rather see more bool params to this function which make the logic more clear.